server/src/main/java/org/elasticsearch/search/aggregations/bucket/filter/FiltersAggregator.java

```diff
@@ -354,6 +361,10 @@ protected LeafBucketCollector getLeafCollector(LeafReaderContext ctx, LeafBucket
                 weights = buildWeights(topLevelQuery(), filters);
             }
             Bits live = ctx.reader().getLiveDocs();
+            Counter counter = new Counter(docCountProvider);
+            if (false == docCountProvider.alwaysOne()) {
+                segmentsWithDocCount++;
+            }
```

Comment:
It's possible to use a different `Counter` implementation that always returns 1 when the provider will always return one. I trust the jvm to inline the calls on it regardless because `counter` is effectively final. It'd save a couple of comparisons in this very very hot section. But performance tests seemed fine so I didn't end up doing it.